proposals/0000-local-import.rst

```diff
@@ -0,0 +1,159 @@
+
+To avoid polluting the namespace with conflicting identifiers, Haskell programmers can choose between explicit and qualified imports ; however, they are often reluctant to do so as:
+- Explicit import lists are inconvenient (they must be edited whenever the code changes). 
+- Module qualifiers can severely affect readability.
+
+The latter problem is often addressed by choosing one-letter qualifiers, which can only be understood by looking up their definition at the top of the file. It is an imperfect solution at best, and does not work well for operators ; for this reason, operators are frequently overloaded via lawless typeclasses for mere syntactic convenience.
+
+OCaml 3.12, released in 2011, introduced two syntactic contructs for locally opening modules: ``let open Module in …`` and ``Module.( … )``. In less than 10 years, this feature has gained widespread adoption throughout the ecosystem, bringing significant benefits in terms of readability and maintainability. For the reasons stated above, we believe Haskell should follow its example.
+
+Proposed Change Specification
+-----------------------------
+We allow qualified and non-qualified ``import`` statements at the beginning of any ``let``/``where``/``do``-block.
+
+For all constructs, the effect of such imports is limited to the scope the enclosing block. For ``let`` and ``where``, the imports also affect the corresponding expression. For example:
+::
+
+  let import Foo in <expr>
+allows the use of symbols from ``Foo`` in the ``<expr>``. Similary, the same thing is possible with:
+::
+
+  f = <expr>
+    where import Foo
+
+Moreover, ``import`` statements are allowed to refer to any module qualifier specified outside that block (that is, local ``import`` statements are not limited to full module names). This means that the following is allowed:
+::
+
+  import qualified Data.Set as Set
+  foo = …
+    where import Set
+          …
+As well as:
+::
+
+  foo = …
+    where import qualified Data.Set as Set
+          bar = …
+            where import Set
+                  …
+But the following is not valid:
+::
+
+  foo = …
+    where import qualified Data.Set as Set
+          import Set
+This is consistent with today's semantics for ``import``, which does not allow for:
+::
+
+  module MyModule where
+
+  import qualified Data.Set as Set
+  import Set
+
+Finally, we allow the syntactic shortcut ``Qualifier.{ <expression> }``, which simply desugars to:
+::
+
+  let import Qualifier in <expression>
+
+The following changes in the Haskell 2010 grammar are required:
+
+::
+
+  decls → { impdecls ; decl_1 ; … ; decl_n }    (n ≥ 0)
+       | { decl_1 ; … ; decl_n }
+  lexp → …
+       | modid.{ lexp }
+       | do { stmts }
+       | do { impdecls ; stmts }
+
+
+Effect and Interactions
+-----------------------
+Detail how the proposed change addresses the original problem raised in the motivation.
```

Comment:
Put emphasis here on the fact that it does not change the current semantic, just add more feature.